crates/types/src/schema/metadata/updater/mod.rs

```diff
@@ -354,29 +355,39 @@ impl SchemaUpdater {
             .collect::<Result<HashMap<_, _>, _>>()?;
 
         // Did we find an existing deployment with a conflicting endpoint url?
-        let mut existing_deployments = self.schema.deployments.iter().filter(|(_, schemas)| {
-            schemas.ty.protocol_type() == deployment_metadata.ty.protocol_type()
-                && schemas.ty.normalized_address() == deployment_metadata.ty.normalized_address()
-        });
+        let existing_deployment = self
+            .schema
+            .deployments
+            .iter()
+            .filter(|(_, schemas)| {
+                schemas.ty.protocol_type() == deployment_metadata.ty.protocol_type()
+                    && schemas.ty.normalized_address()
+                        == deployment_metadata.ty.normalized_address()
+                    && routing_header.as_ref().is_none_or(
```

Comment:
This means if a new deployment that has No routing header will conflict with a deployment that has a routing header? 

is it possible that a single deployment (behind the LB) is routed to as "default" without any headers and multiple ones with a routing header?

If this is true, this means conflicting here is not correct.

Reply:
What will happen if I did this instead:
- localhost:9080 without routing header (works)
- localhost:9080 with routing header (will also work) will not conflict. 

So the system will behave differently by changing the order of registration. 

What i think might work instead is to filter mainly like `address == deployment.address && routing_header == deployment.routing_header`